File: cpp/2287. 重排字符形成目标字符串.cpp
class Solution {
public:
    int rearrangeCharacters(string s, string target) {
        unordered_map<char,int> sCnt,targetCnt;
        for(auto &ch:target){
            targetCnt[ch]++;
        }
        for(auto &ch:s){
            sCnt[ch]++;
        }
        int ans=INT_MAX;
        for(auto&[c,cnt]:targetCnt){
            ans=min(ans,sCnt[c]/cnt);
            if(ans==0) return 0;
        }
        return ans;
    }
};

File: 刷题/golang/2287. 重排字符形成目标字符串.go
func rearrangeCharacters(s string, target string) int {
    var sCnt,targetCnt [26]int
    for _,c:=range s{
        sCnt[c-'a']++
    }
    for _,c:=range target{
        targetCnt[c-'a']++
    }
    ans:=len(s)
    for i,c:=range targetCnt{
        if c>0{
            ans=min(ans,sCnt[i]/c)
        }
        
        if ans==0{
            return 0
        }
    }
    return ans
}
func min(a,b int) int{
    if a<b{
        return a
    }
    return b
}